harness/determined/cli/job.py

```diff
@@ -0,0 +1,106 @@
+import json
+from argparse import Namespace
+from typing import Any, List
+
+import yaml
+
+from determined.cli import render
+from determined.cli.session import setup_session
+from determined.common.api import authentication
+from determined.common.api.b import get_GetJobs
+from determined.common.declarative_argparse import Arg, Cmd
+
+
+@authentication.required
+def ls(args: Namespace) -> None:
+    response = get_GetJobs(
+        setup_session(args),
+        resourcePool=args.resource_pool,
+        pagination_limit=args.limit,
+        pagination_offset=args.offset,
+        orderBy=args.order_by,
+    )
+    if args.output == "yaml":
+        print(yaml.safe_dump(response.to_json(), default_flow_style=False))
+    elif args.output == "json":
+        print(json.dumps(response.to_json(), indent=4, default=str))
+    elif ["csv", "table"].count(args.output) > 0:
+        headers = [
+            "Jobs Ahead",
+            "ID",
+            "Entity ID",
+            "Status",
+            "Type",
+            "Slots Acquired",
+            "Slots Requested",
+            "Name",
+            "User",
+            "Submission Time",
+        ]
+        values = [
+            [
+                j.summary.jobsAhead
+                if j.summary is not None and j.summary.jobsAhead > -1
+                else "N/A",
+                j.jobId,
+                j.entityId,
+                j.summary.state if j.summary is not None else "N/A",
+                j.type,
+                j.allocatedSlots,
+                j.requestedSlots,
+                j.name,
+                j.username,
+                j.submissionTime,
+            ]
+            for j in response.jobs
+        ]
+        render.tabulate_or_csv(headers, values, as_csv=args.output == "csv")
+    else:
+        raise ValueError(f"Bad output format: {args.output}")
+
+
+pagination_args = [
+    Arg(
+        "--offset",
+        type=int,
+        default=0,
+        help="Offset the returned set.",
+    ),
+    Arg(
+        "--limit",
+        type=int,
+        default=50,
+        help="Limit the returned set.",
+    ),
+    Arg(
+        "--order-by",
+        type=str,
+        help="Whether to sort the results in descending or ascending order",
+    ),
+]
+
+args_description = [
+    Cmd(
+        "j|ob",
+        None,
+        "manage job",
+        [
+            Cmd(
+                "list",
+                ls,
+                "list jobs",
+                [
+                    Arg(
+                        "-o",
+                        "--output",
```

Comment:
TODO make sure thiis style is consistent with the rest of the cli. this arg should be somethin we can share across different cli commands like pagination.